src/libraries/Microsoft.PowerFx.Core/Syntax/Nodes/DottedNameNode.cs

```diff
@@ -12,28 +12,38 @@
 
 namespace Microsoft.PowerFx.Core.Syntax.Nodes
 {
-    internal sealed class DottedNameNode : NameNode
+    public sealed class DottedNameNode : NameNode
     {
-        public readonly TexlNode Left;
-        public readonly Identifier Right;
+        public TexlNode Left { get; }
+
+        public Identifier Right { get; }
 
         // Can be null
-        public readonly TexlNode RightNode;
-        public readonly bool HasOnlyIdentifiers;
-        public readonly bool HasPossibleNamespaceQualifier;
+        internal readonly TexlNode RightNode;
+
+        public bool HasOnlyIdentifiers { get; }
+
+        public bool HasPossibleNamespaceQualifier { get; }
 
+        /// <inheritdoc />
         public override NodeKind Kind => NodeKind.DottedName;
 
-        // True if the name uses dots, e.g. A.B.C
+        /// <summary>
+        /// True if the name uses dots, e.g. A.B.C.
+        /// </summary>
         public bool UsesDot => Token.Kind == TokKind.Dot;
 
-        // True if the name uses bangs, e.g. A!B!C
+        /// <summary>
+        /// True if the name uses bangs, e.g. A!B!C.
+        /// </summary>
         public bool UsesBang => Token.Kind == TokKind.Bang;
 
-        // True if the name uses brackets, e.g. A[B]
+        /// <summary>
+        /// True if the name uses brackets, e.g. A[B].
+        /// </summary>
         public bool UsesBracket => Token.Kind == TokKind.BracketOpen;
```

Comment:
What is this feature? Maybe keep internal until we understand how it's used or if it's removable?

Reply:
Same for HasOnlyIdentifiers/HasPossibleNamespaceQualifier